### PR TITLE
CBG-2551 Collection access for _user REST API

### PR DESCRIPTION
**Pull request by adamcfraser**
CBG-2551

Adds basic read/write of collection explicit channels for /db/_user endpoint.

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1197/

### CODE REVIEW

File: auth/principal.go [hunk 1]
```diff
@@ -148,8 +148,9 @@ type User interface {
 // Used to define a user/role within DbConfig, and structures the request/response body in the admin REST API
 // for /db/_user/*
 type PrincipalConfig struct {
-	Name             *string  `json:"name,omitempty"`
-	ExplicitChannels base.Set `json:"admin_channels,omitempty"`
+	Name             *string                                       `json:"name,omitempty"`
+	ExplicitChannels base.Set                                      `json:"admin_channels,omitempty"`
+	CollectionAccess map[string]map[string]*CollectionAccessConfig `json:"collection_access"`
```
Comment (bbrks) [inline auth/principal.go]: Should we add `omitempty` to reduce storage overhead for non-collection deployments?

```suggestion
	CollectionAccess map[string]map[string]*CollectionAccessConfig `json:"collection_access,omitempty"`
```

File: rest/user_api_test.go [hunk 2]
```diff
@@ -1352,3 +1352,59 @@ func TestUserXattrAvoidRevisionIDGeneration(t *testing.T) {
...
+	RequireStatus(t, getResponse, 200)
+	log.Printf("response:%s", getResponse.Body.Bytes())
+	var responseConfig auth.PrincipalConfig
+	err := json.Unmarshal(getResponse.Body.Bytes(), &responseConfig)
+	require.NoError(t, err)
+	collectionAccess, ok := responseConfig.CollectionAccess["scope1"]["collection1"]
+	require.True(t, ok)
+	assert.Equal(t, channels.BaseSetOf(t, "foo", "bar1"), collectionAccess.ExplicitChannels_)
+	// TODO: computed channels requires authenticator populated with collection set, pending CBG-2266
+	//assert.Equal(t, channels.BaseSetOf(t), collectionAccess.Channels_)
```
Comment (bbrks) [inline rest/user_api_test.go:1388]: Is this now pending RestTester collections? And does this (and the below test cases) make sense to spin off?

**Review by bbrks — DISMISSED** [review]
Some comments that can come later as follow-ups, but happy to merge to get the API aligned.

**Review by bbrks — APPROVED** [review]

File: rest/user_api_test.go (hunk 2 above)
Comment (adamcfraser) [reply rest/user_api_test.go:1388]: Good idea - filed CBG-2616 for this.

**Pull request merged**